frontend/hub/api.tsx

```diff
@@ -0,0 +1,103 @@
+import { AutomationServerType } from '../automation-servers/AutomationServer';
+import { activeAutomationServer } from '../automation-servers/AutomationServersProvider';
+
+function apiTag(strings: TemplateStringsArray, ...values: string[]) {
+  if (strings[0]?.[0] !== '/') {
+    throw new Error('Invalid URL');
+  }
+
+  let url = '';
+  strings.forEach((fragment, index) => {
+    url += fragment;
+    if (index !== strings.length - 1) {
+      url += encodeURIComponent(`${values.shift() ?? ''}`);
+    }
+  });
+
+  return url;
+}
+
+export function hubAPI(strings: TemplateStringsArray, ...values: string[]) {
+  let base = process.env.HUB_API_BASE_PATH;
+  if (!base) {
+    if (activeAutomationServer?.type === AutomationServerType.Galaxy) {
```

Comment:
(remember to rebase, this file already exists on main, and the activeAutomationServer logic is obsolete)